### PR TITLE
T/refactor composer

### PR DESCRIPTION
**Pull request by till**
Related:
easybib/issues#4121
easybib/issues#4112

### CODE REVIEW

File: easybib/recipes/role-phpapp.rb [hunk 1]
```diff
@@ -4,7 +4,10 @@
 include_recipe 'php-phar'
 include_recipe 'php-suhosin'
 include_recipe 'php-zlib'
+
+node.set['composer']['environment'] = get_deploy_user
```
Comment (till) [inline easybib/recipes/role-phpapp.rb]: @fh What do you think? For the record, I did not test deploy this yet.

File: composer/recipes/configure.rb [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
-  case deploy_user['user']
-  when 'root'
-    home_dir = '/root'
-  when 'www-data'
-    home_dir = '/var/www'
-  else
-    home_dir = "/home/#{deploy_user['user']}"
-  end
-end
+return if deploy_user['user'].nil?
```
Comment (fh) [inline composer/recipes/configure.rb]: I would add a log statement here

File: easybib/recipes/role-phpapp.rb (hunk 1 above)
Comment (fh) [inline easybib/recipes/role-phpapp.rb]: lgtm, lets just try this on playground before merging to stable

File: composer/recipes/configure.rb (hunk 2 above)
Comment (till) [inline composer/recipes/configure.rb]: @fh info?

**Pull request merged**